### PR TITLE
Fix merge-index-image bugs

### PR DESCRIPTION
**Pull request by yashvardhannanavati**

### CODE REVIEW

File: iib/workers/tasks/utils.py [hunk 1]
```diff
@@ -814,18 +814,20 @@ def prepare_request_for_build(request_id, build_request_config):
...
+        request_index = index_info['target_index']
+
+    resolved_distribution_scope = request_index['resolved_distribution_scope']
 
     distribution_scope = _validate_distribution_scope(
         resolved_distribution_scope, build_request_config.distribution_scope
     )
 
-    binary_image = build_request_config.binary_image(index_info['from_index'], distribution_scope)
+    binary_image = build_request_config.binary_image(request_index, distribution_scope)
```
Comment (lcarva) [inline iib/workers/tasks/utils.py:830]: It would be nice to have a unit test to ensure this regression doesn't happen again

**Review by gallettilance — DISMISSED** [review]

File: iib/workers/tasks/utils.py (hunk 1 above)
Comment (yashvardhannanavati) [reply iib/workers/tasks/utils.py:830]: The tests were there but they were by chance asserting the wrong info. I tweaked the test to fail if things change.

File: iib/workers/tasks/build_merge_index_image.py [hunk 2]
```diff
@@ -246,6 +246,20 @@ def handle_merge_request(
                 overwrite_target_index_token,
             )
 
+        _add_label_to_index(
```
Comment (pbortlov) [inline iib/workers/tasks/build_merge_index_image.py:249]: Do you think we could loop over those to labels?

labels = {'com.redhat.index.delivery.version': prebuild_info['target_ocp_version'], 'com.redhat.index.delivery.distribution_scope': prebuild_info['distribution_scope']}
for label,value in labels.items():
   _add_label_to_index(label,value, temp_dir, 'index.Dockerfile')

File: tests/test_workers/test_tasks/test_utils.py [hunk 3]
```diff
@@ -864,9 +864,9 @@ def test_prepare_request_for_build_merge_index_img(mock_gia, mock_gri, mock_giii
 
     target_index_info = {
         'resolved_from_index': 'some_other_image@sha256',
-        'ocp_version': 'v4.6',
+        'ocp_version': 'v4.9',
         'arches': {'amd64'},
-        'resolved_distribution_scope': 'stage',
+        'resolved_distribution_scope': 'prod',
```
Comment (pbortlov) [inline tests/test_workers/test_tasks/test_utils.py:869]: Why is this changed?

File: iib/workers/tasks/build_merge_index_image.py (hunk 2 above)
Comment (yashvardhannanavati) [reply iib/workers/tasks/build_merge_index_image.py:249]: I'm just trying to keep it consistent with the `Add` request implementation. Please let me know if you feel strongly against it

File: tests/test_workers/test_tasks/test_utils.py (hunk 3 above)
Comment (yashvardhannanavati) [reply tests/test_workers/test_tasks/test_utils.py:869]: To make sure the tests break if something changes in `prepare_request_for_build`

**Review by lcarva — APPROVED** [review]

File: iib/workers/tasks/build_merge_index_image.py (hunk 2 above)
Comment (pbortlov) [reply iib/workers/tasks/build_merge_index_image.py:249]: well in create-empty-index I try this approach.
